src/auth/jwtclient.ts

```diff
@@ -44,15 +52,22 @@ export class JWT extends OAuth2Client {
    * @param {string=} subject impersonated account's email address.
    * @constructor
    */
+  constructor(options: JWTOptions);
+  constructor(
+      email?: string, keyFile?: string, key?: string, scopes?: string|string[],
+      subject?: string);
   constructor(
-      email?: string, keyFile?: string|null, key?: string|null,
-      scopes?: string|string[]|null, subject?: string) {
+      optionsOrEmail?: string|JWTOptions, keyFile?: string, key?: string,
+      scopes?: string|string[], subject?: string) {
     super();
-    this.email = email;
-    this.keyFile = keyFile;
-    this.key = key;
-    this.scopes = scopes;
-    this.subject = subject;
+    const opts = (optionsOrEmail && typeof optionsOrEmail === 'object') ?
+        optionsOrEmail :
+        {email: optionsOrEmail, keyFile, key, scopes, subject};
+    this.email = opts.email;
+    this.keyFile = opts.keyFile;
+    this.key = opts.key;
+    this.scopes = opts.scopes;
+    this.subject = opts.subject;
```

Comment:
I was thinking that there perhaps there is a clever way to employ [destructuring](https://stackoverflow.com/questions/17781472/how-to-get-a-subset-of-a-javascript-objects-properties) here. The result is overcomplicated. Worst of all it repeats property names.

```js
Object.assign(this, 
  (({email, keyFile, key, scopes, subject}) => 
   ({email, keyFile, key, scopes, subject}))(opts));
```

An alternatively would be `lodash.pick`: 
```js
Object.assign(this, pick(opts, ['email', 'keyFile', 'key', 'scopes', 'subject']));
```

On second thought; I think your version is fine. Fun exercise though.

Reply:
I had the same exact thought :)